File: elixir/problem4.exs
defmodule Palendrome do

  def check_pal(num) do
    num_to_string = Integer.to_string(num)
    is_pal(num_to_string)
  end

  defp is_pal(num) do
    num == String.reverse(num) 
  end

end


defmodule DigitMult do
  
  def get_nums(num_digits) do
   start_num = 9 * num_digits
   IO.puts gen_nums(start_num, start_num, start_num)
  end

  def gen_nums(num1, num2, start_num) do
    IO.puts num1
    IO.puts num2
    product = num1 * num2
    IO.puts product
    if Palendrome.check_pal(product) do
      product
    else
      new_num1 = num1
      new_num2 = num2
      if rem(num2, 100) == 0 do
        new_num1 = num1 - 1
        num2 = start_num
      end
      gen_nums(new_num1, num2 - 1, start_num)
    end
  end
end

DigitMult.get_nums(111)

File: js/problem4.js

function isPal(num) {
  var reversed_string = reverse(num.toString())
  return num == Number(reversed_string);
}

function reverse(string) {
  return string.split('').reverse().join('');
}

function generateNum(digit){
  var array = [];
  for(var i = 0; i < digit; i++) {
    array.push(9);
  }
  console.log(array);
  return Number(array.join(""));
}

function generateMod(digit){
  var array = [];
  for(var i = 0; i < digit; i++) {
    if(i == 0) {
      array.push(1);
    } else {
      array.push(0);
    }
  }
  console.log(array);
  return Number(array.join(""));
}


function getNums(digit) {
  var startNum = generateNum(digit);
  var modNum = generateMod(digit);
  var pal = find_pal(startNum, startNum, startNum, modNum)
  console.log(pal);
}

function find_pal(num1, num2, startNum, modNum) {
  var product = num1 * num2;
  if (isPal(product)){
    return product
  } else {
    var new1 = num2 % modNum === 0 ? num1 - 1 : num1;
    var new2 = num2 % modNum === 0 ? startNum : num2 - 1;
    return find_pal(new1, new2, startNum, modNum);
  }
}

getNums(3);


